### ISSUE TITLE
Check and document reproducible builds

### ISSUE THREAD

**Issue by tsusanka** [trezor/trezor-firmware#692]
Part of #685.

Check or add a guide how to build firmware yourself and check it against the one provided by us. Some content is on our wiki https://wiki.trezor.io/Developers_guide:Deterministic_firmware_build but it is outdated.

I suggest to document it here in this repository and add a link to the wiki.

**Comment by prusnak** [comment]
I just checked the wiki and it seems pretty fine - i.e. there is no outdated stuff there.

**Comment by tsusanka** [comment]
I have tried this for 2.3.1 and the fingerprints did not match. So this is most likely caused due to the fact that we have introduced NixOS after 2.3.1. So let me try again during the next release and let's close then.

**Comment by mmilata** [comment]
I've reviewed https://wiki.trezor.io/Developers_guide:Deterministic_firmware_build and it seems correct, at least for recent versions.

* `legacy/v1.9.0` + `core/v2.3.0`: build gets stuck due to micropython submodule that fell off github
* `legacy/v1.9.4` + `core/v2.3.5`: first version with newer micropython, builds fine, comparison to official fw OK
* `legacy/v1.10.3` + `core/v2.4.2`: builds fine, comparison OK

Verifying legacy FW is a bit complicated due to the legacy header. #1918 attempts to improve the description of the process.

**Comment by bosomt** [comment]
Does it needs QA or i can move it to approved issues section ?

**Comment by tsusanka** [comment]
Let's approve, Martin basically tested himself :).
